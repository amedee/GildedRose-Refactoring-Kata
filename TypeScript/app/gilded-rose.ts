import {CustomItemFactory} from "./items";

export class Item {
    name: string;
    sellIn: number;
    quality: number;

    constructor(name, sellIn, quality) {
        this.name = name;
        this.sellIn = sellIn;
        this.quality = quality;
    }
}

export class GildedRose {
    itemFactory: CustomItemFactory;
    items: Array<Item>;

    constructor(items = [] as Array<Item>) {
        this.itemFactory = new CustomItemFactory();
        this.items = items;
    }

    updateQuality() {
        this.items.forEach(item => {
            this.itemFactory.customiseItem(item).updateState();
            this.checkQualityRange(item);
        });
        return this.items;
    }

    private checkQualityRange(item: Item) {
        if (item.name != 'Sulfuras, Hand of Ragnaros') {
            item.quality = Math.max(Math.min(item.quality, 50), 0);
        }
    }
}
