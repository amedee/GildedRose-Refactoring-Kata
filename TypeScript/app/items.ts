import {Item} from "./gilded-rose";

export interface CustomItem {
    updateState(): void;
}

export class AgedBrie implements CustomItem {
    private item: Item;

    constructor(item: Item) {
        this.item = item;
    }

    updateState(): void {
        this.item.sellIn -= 1;
        if (this.item.sellIn < 0) {
            this.item.quality += 2;
        } else {
            this.item.quality += 1;
        }
    }
}

class BackstagePass implements CustomItem {
    private item: Item;

    constructor(item: Item) {
        this.item = item;
    }

    updateState(): void {
        this.item.sellIn -= 1;
        if (this.item.sellIn < 0) {
            this.item.quality = 0;
        } else if (this.item.sellIn < 5) {
            this.item.quality += 3;
        } else if (this.item.sellIn < 10) {
            this.item.quality += 2;
        } else {
            this.item.quality += 1;
        }
    }
}

class Sulfuras implements CustomItem {
    private item: Item;

    constructor(item: Item) {
        this.item = item;
    }

    updateState(): void {
        // Nothing to do!
    }
}

class StandardItem implements CustomItem {
    private item: Item;

    constructor(item: Item) {
        this.item = item;
    }

    updateState(): void {
        this.item.sellIn -= 1;
        if (this.item.sellIn < 0) {
            this.item.quality += -2;
        } else {
            this.item.quality += -1;
        }
    }
}

export class CustomItemFactory {
    constructor() {
    }

    private readonly AGED_BRIE = 'Aged Brie';
    private readonly BACKSTAGE_PASS = 'Backstage passes to a TAFKAL80ETC concert';
    private readonly SULFURAS = 'Sulfuras, Hand of Ragnaros';

    customiseItem(item: Item): CustomItem {
        switch (item.name) {
            case this.AGED_BRIE:
                return new AgedBrie(item);
            case this.BACKSTAGE_PASS:
                return new BackstagePass(item);
            case this.SULFURAS:
                return new Sulfuras(item);
            default:
                return new StandardItem(item);
        }
    }
}
