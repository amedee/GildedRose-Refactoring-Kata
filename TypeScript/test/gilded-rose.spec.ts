import {GildedRose, Item} from '../app/gilded-rose';
import {goldenMaster} from "./golden-master-text-test";

function doUpdateQuality(name: string, sellIn: number, quality: number) {
    const gildedRose = new GildedRose([new Item(name, sellIn, quality)]);
    const items = gildedRose.updateQuality();
    return JSON.stringify(items[0]);
}

describe('Gilded Rose', () => {

    it('should update quality', () =>
        expect(doUpdateQuality).toVerifyAllCombinations(
            [
                'foo'
            ],
            [-1],
            [0]
        ));

    it('should be verified with golden master', () =>
        expect(goldenMaster()).toMatchSnapshot());
});
