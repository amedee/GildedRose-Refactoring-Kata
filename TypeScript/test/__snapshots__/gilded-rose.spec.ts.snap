// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gilded Rose should be verified with golden master 1`] = `
"-------- day 0 --------
name, sellIn, quality
+5 Dexterity Vest 10 20
Aged Brie 2 0
Elixir of the Mongoose 5 7
Sulfuras, Hand of Ragnaros 0 80
Sulfuras, Hand of Ragnaros -1 80
Backstage passes to a TAFKAL80ETC concert 15 20
Backstage passes to a TAFKAL80ETC concert 10 49
Backstage passes to a TAFKAL80ETC concert 5 49
Conjured Mana Cake 3 6

-------- day 1 --------
name, sellIn, quality
+5 Dexterity Vest 9 19
Aged Brie 1 1
Elixir of the Mongoose 4 6
Sulfuras, Hand of Ragnaros 0 80
Sulfuras, Hand of Ragnaros -1 80
Backstage passes to a TAFKAL80ETC concert 14 21
Backstage passes to a TAFKAL80ETC concert 9 50
Backstage passes to a TAFKAL80ETC concert 4 50
Conjured Mana Cake 2 4

"
`;

exports[`Gilded Rose should update quality 1`] = `
Object {
  "Aged Brie,-1,0": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":-2,\\"quality\\":2}",
  "Aged Brie,-1,1": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":-2,\\"quality\\":3}",
  "Aged Brie,-1,49": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":-2,\\"quality\\":50}",
  "Aged Brie,-1,50": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":-2,\\"quality\\":50}",
  "Aged Brie,0,0": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":-1,\\"quality\\":2}",
  "Aged Brie,0,1": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":-1,\\"quality\\":3}",
  "Aged Brie,0,49": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":-1,\\"quality\\":50}",
  "Aged Brie,0,50": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":-1,\\"quality\\":50}",
  "Aged Brie,1,0": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":0,\\"quality\\":1}",
  "Aged Brie,1,1": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":0,\\"quality\\":2}",
  "Aged Brie,1,49": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":0,\\"quality\\":50}",
  "Aged Brie,1,50": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":0,\\"quality\\":50}",
  "Aged Brie,10,0": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":9,\\"quality\\":1}",
  "Aged Brie,10,1": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":9,\\"quality\\":2}",
  "Aged Brie,10,49": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":9,\\"quality\\":50}",
  "Aged Brie,10,50": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":9,\\"quality\\":50}",
  "Aged Brie,11,0": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":10,\\"quality\\":1}",
  "Aged Brie,11,1": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":10,\\"quality\\":2}",
  "Aged Brie,11,49": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":10,\\"quality\\":50}",
  "Aged Brie,11,50": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":10,\\"quality\\":50}",
  "Aged Brie,12,0": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":11,\\"quality\\":1}",
  "Aged Brie,12,1": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":11,\\"quality\\":2}",
  "Aged Brie,12,49": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":11,\\"quality\\":50}",
  "Aged Brie,12,50": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":11,\\"quality\\":50}",
  "Aged Brie,5,0": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":4,\\"quality\\":1}",
  "Aged Brie,5,1": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":4,\\"quality\\":2}",
  "Aged Brie,5,49": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":4,\\"quality\\":50}",
  "Aged Brie,5,50": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":4,\\"quality\\":50}",
  "Aged Brie,6,0": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":5,\\"quality\\":1}",
  "Aged Brie,6,1": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":5,\\"quality\\":2}",
  "Aged Brie,6,49": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":5,\\"quality\\":50}",
  "Aged Brie,6,50": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":5,\\"quality\\":50}",
  "Aged Brie,7,0": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":6,\\"quality\\":1}",
  "Aged Brie,7,1": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":6,\\"quality\\":2}",
  "Aged Brie,7,49": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":6,\\"quality\\":50}",
  "Aged Brie,7,50": "{\\"name\\":\\"Aged Brie\\",\\"sellIn\\":6,\\"quality\\":50}",
  "Backstage passes to a TAFKAL80ETC concert,-1,0": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":-2,\\"quality\\":0}",
  "Backstage passes to a TAFKAL80ETC concert,-1,1": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":-2,\\"quality\\":0}",
  "Backstage passes to a TAFKAL80ETC concert,-1,49": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":-2,\\"quality\\":0}",
  "Backstage passes to a TAFKAL80ETC concert,-1,50": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":-2,\\"quality\\":0}",
  "Backstage passes to a TAFKAL80ETC concert,0,0": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":-1,\\"quality\\":0}",
  "Backstage passes to a TAFKAL80ETC concert,0,1": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":-1,\\"quality\\":0}",
  "Backstage passes to a TAFKAL80ETC concert,0,49": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":-1,\\"quality\\":0}",
  "Backstage passes to a TAFKAL80ETC concert,0,50": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":-1,\\"quality\\":0}",
  "Backstage passes to a TAFKAL80ETC concert,1,0": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":0,\\"quality\\":3}",
  "Backstage passes to a TAFKAL80ETC concert,1,1": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":0,\\"quality\\":4}",
  "Backstage passes to a TAFKAL80ETC concert,1,49": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":0,\\"quality\\":50}",
  "Backstage passes to a TAFKAL80ETC concert,1,50": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":0,\\"quality\\":50}",
  "Backstage passes to a TAFKAL80ETC concert,10,0": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":9,\\"quality\\":2}",
  "Backstage passes to a TAFKAL80ETC concert,10,1": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":9,\\"quality\\":3}",
  "Backstage passes to a TAFKAL80ETC concert,10,49": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":9,\\"quality\\":50}",
  "Backstage passes to a TAFKAL80ETC concert,10,50": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":9,\\"quality\\":50}",
  "Backstage passes to a TAFKAL80ETC concert,11,0": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":10,\\"quality\\":1}",
  "Backstage passes to a TAFKAL80ETC concert,11,1": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":10,\\"quality\\":2}",
  "Backstage passes to a TAFKAL80ETC concert,11,49": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":10,\\"quality\\":50}",
  "Backstage passes to a TAFKAL80ETC concert,11,50": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":10,\\"quality\\":50}",
  "Backstage passes to a TAFKAL80ETC concert,12,0": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":11,\\"quality\\":1}",
  "Backstage passes to a TAFKAL80ETC concert,12,1": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":11,\\"quality\\":2}",
  "Backstage passes to a TAFKAL80ETC concert,12,49": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":11,\\"quality\\":50}",
  "Backstage passes to a TAFKAL80ETC concert,12,50": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":11,\\"quality\\":50}",
  "Backstage passes to a TAFKAL80ETC concert,5,0": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":4,\\"quality\\":3}",
  "Backstage passes to a TAFKAL80ETC concert,5,1": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":4,\\"quality\\":4}",
  "Backstage passes to a TAFKAL80ETC concert,5,49": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":4,\\"quality\\":50}",
  "Backstage passes to a TAFKAL80ETC concert,5,50": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":4,\\"quality\\":50}",
  "Backstage passes to a TAFKAL80ETC concert,6,0": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":5,\\"quality\\":2}",
  "Backstage passes to a TAFKAL80ETC concert,6,1": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":5,\\"quality\\":3}",
  "Backstage passes to a TAFKAL80ETC concert,6,49": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":5,\\"quality\\":50}",
  "Backstage passes to a TAFKAL80ETC concert,6,50": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":5,\\"quality\\":50}",
  "Backstage passes to a TAFKAL80ETC concert,7,0": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":6,\\"quality\\":2}",
  "Backstage passes to a TAFKAL80ETC concert,7,1": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":6,\\"quality\\":3}",
  "Backstage passes to a TAFKAL80ETC concert,7,49": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":6,\\"quality\\":50}",
  "Backstage passes to a TAFKAL80ETC concert,7,50": "{\\"name\\":\\"Backstage passes to a TAFKAL80ETC concert\\",\\"sellIn\\":6,\\"quality\\":50}",
  "Conjured Mana Cake,-1,0": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":-2,\\"quality\\":0}",
  "Conjured Mana Cake,-1,1": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":-2,\\"quality\\":0}",
  "Conjured Mana Cake,-1,49": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":-2,\\"quality\\":45}",
  "Conjured Mana Cake,-1,50": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":-2,\\"quality\\":46}",
  "Conjured Mana Cake,0,0": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":-1,\\"quality\\":0}",
  "Conjured Mana Cake,0,1": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":-1,\\"quality\\":0}",
  "Conjured Mana Cake,0,49": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":-1,\\"quality\\":45}",
  "Conjured Mana Cake,0,50": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":-1,\\"quality\\":46}",
  "Conjured Mana Cake,1,0": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":0,\\"quality\\":0}",
  "Conjured Mana Cake,1,1": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":0,\\"quality\\":0}",
  "Conjured Mana Cake,1,49": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":0,\\"quality\\":47}",
  "Conjured Mana Cake,1,50": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":0,\\"quality\\":48}",
  "Conjured Mana Cake,10,0": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":9,\\"quality\\":0}",
  "Conjured Mana Cake,10,1": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":9,\\"quality\\":0}",
  "Conjured Mana Cake,10,49": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":9,\\"quality\\":47}",
  "Conjured Mana Cake,10,50": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":9,\\"quality\\":48}",
  "Conjured Mana Cake,11,0": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":10,\\"quality\\":0}",
  "Conjured Mana Cake,11,1": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":10,\\"quality\\":0}",
  "Conjured Mana Cake,11,49": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":10,\\"quality\\":47}",
  "Conjured Mana Cake,11,50": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":10,\\"quality\\":48}",
  "Conjured Mana Cake,12,0": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":11,\\"quality\\":0}",
  "Conjured Mana Cake,12,1": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":11,\\"quality\\":0}",
  "Conjured Mana Cake,12,49": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":11,\\"quality\\":47}",
  "Conjured Mana Cake,12,50": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":11,\\"quality\\":48}",
  "Conjured Mana Cake,5,0": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":4,\\"quality\\":0}",
  "Conjured Mana Cake,5,1": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":4,\\"quality\\":0}",
  "Conjured Mana Cake,5,49": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":4,\\"quality\\":47}",
  "Conjured Mana Cake,5,50": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":4,\\"quality\\":48}",
  "Conjured Mana Cake,6,0": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":5,\\"quality\\":0}",
  "Conjured Mana Cake,6,1": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":5,\\"quality\\":0}",
  "Conjured Mana Cake,6,49": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":5,\\"quality\\":47}",
  "Conjured Mana Cake,6,50": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":5,\\"quality\\":48}",
  "Conjured Mana Cake,7,0": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":6,\\"quality\\":0}",
  "Conjured Mana Cake,7,1": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":6,\\"quality\\":0}",
  "Conjured Mana Cake,7,49": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":6,\\"quality\\":47}",
  "Conjured Mana Cake,7,50": "{\\"name\\":\\"Conjured Mana Cake\\",\\"sellIn\\":6,\\"quality\\":48}",
  "Sulfuras, Hand of Ragnaros,-1,0": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":-1,\\"quality\\":0}",
  "Sulfuras, Hand of Ragnaros,-1,1": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":-1,\\"quality\\":1}",
  "Sulfuras, Hand of Ragnaros,-1,49": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":-1,\\"quality\\":49}",
  "Sulfuras, Hand of Ragnaros,-1,50": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":-1,\\"quality\\":50}",
  "Sulfuras, Hand of Ragnaros,0,0": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":0,\\"quality\\":0}",
  "Sulfuras, Hand of Ragnaros,0,1": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":0,\\"quality\\":1}",
  "Sulfuras, Hand of Ragnaros,0,49": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":0,\\"quality\\":49}",
  "Sulfuras, Hand of Ragnaros,0,50": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":0,\\"quality\\":50}",
  "Sulfuras, Hand of Ragnaros,1,0": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":1,\\"quality\\":0}",
  "Sulfuras, Hand of Ragnaros,1,1": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":1,\\"quality\\":1}",
  "Sulfuras, Hand of Ragnaros,1,49": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":1,\\"quality\\":49}",
  "Sulfuras, Hand of Ragnaros,1,50": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":1,\\"quality\\":50}",
  "Sulfuras, Hand of Ragnaros,10,0": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":10,\\"quality\\":0}",
  "Sulfuras, Hand of Ragnaros,10,1": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":10,\\"quality\\":1}",
  "Sulfuras, Hand of Ragnaros,10,49": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":10,\\"quality\\":49}",
  "Sulfuras, Hand of Ragnaros,10,50": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":10,\\"quality\\":50}",
  "Sulfuras, Hand of Ragnaros,11,0": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":11,\\"quality\\":0}",
  "Sulfuras, Hand of Ragnaros,11,1": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":11,\\"quality\\":1}",
  "Sulfuras, Hand of Ragnaros,11,49": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":11,\\"quality\\":49}",
  "Sulfuras, Hand of Ragnaros,11,50": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":11,\\"quality\\":50}",
  "Sulfuras, Hand of Ragnaros,12,0": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":12,\\"quality\\":0}",
  "Sulfuras, Hand of Ragnaros,12,1": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":12,\\"quality\\":1}",
  "Sulfuras, Hand of Ragnaros,12,49": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":12,\\"quality\\":49}",
  "Sulfuras, Hand of Ragnaros,12,50": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":12,\\"quality\\":50}",
  "Sulfuras, Hand of Ragnaros,5,0": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":5,\\"quality\\":0}",
  "Sulfuras, Hand of Ragnaros,5,1": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":5,\\"quality\\":1}",
  "Sulfuras, Hand of Ragnaros,5,49": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":5,\\"quality\\":49}",
  "Sulfuras, Hand of Ragnaros,5,50": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":5,\\"quality\\":50}",
  "Sulfuras, Hand of Ragnaros,6,0": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":6,\\"quality\\":0}",
  "Sulfuras, Hand of Ragnaros,6,1": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":6,\\"quality\\":1}",
  "Sulfuras, Hand of Ragnaros,6,49": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":6,\\"quality\\":49}",
  "Sulfuras, Hand of Ragnaros,6,50": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":6,\\"quality\\":50}",
  "Sulfuras, Hand of Ragnaros,7,0": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":7,\\"quality\\":0}",
  "Sulfuras, Hand of Ragnaros,7,1": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":7,\\"quality\\":1}",
  "Sulfuras, Hand of Ragnaros,7,49": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":7,\\"quality\\":49}",
  "Sulfuras, Hand of Ragnaros,7,50": "{\\"name\\":\\"Sulfuras, Hand of Ragnaros\\",\\"sellIn\\":7,\\"quality\\":50}",
  "foo,-1,0": "{\\"name\\":\\"foo\\",\\"sellIn\\":-2,\\"quality\\":0}",
  "foo,-1,1": "{\\"name\\":\\"foo\\",\\"sellIn\\":-2,\\"quality\\":0}",
  "foo,-1,49": "{\\"name\\":\\"foo\\",\\"sellIn\\":-2,\\"quality\\":47}",
  "foo,-1,50": "{\\"name\\":\\"foo\\",\\"sellIn\\":-2,\\"quality\\":48}",
  "foo,0,0": "{\\"name\\":\\"foo\\",\\"sellIn\\":-1,\\"quality\\":0}",
  "foo,0,1": "{\\"name\\":\\"foo\\",\\"sellIn\\":-1,\\"quality\\":0}",
  "foo,0,49": "{\\"name\\":\\"foo\\",\\"sellIn\\":-1,\\"quality\\":47}",
  "foo,0,50": "{\\"name\\":\\"foo\\",\\"sellIn\\":-1,\\"quality\\":48}",
  "foo,1,0": "{\\"name\\":\\"foo\\",\\"sellIn\\":0,\\"quality\\":0}",
  "foo,1,1": "{\\"name\\":\\"foo\\",\\"sellIn\\":0,\\"quality\\":0}",
  "foo,1,49": "{\\"name\\":\\"foo\\",\\"sellIn\\":0,\\"quality\\":48}",
  "foo,1,50": "{\\"name\\":\\"foo\\",\\"sellIn\\":0,\\"quality\\":49}",
  "foo,10,0": "{\\"name\\":\\"foo\\",\\"sellIn\\":9,\\"quality\\":0}",
  "foo,10,1": "{\\"name\\":\\"foo\\",\\"sellIn\\":9,\\"quality\\":0}",
  "foo,10,49": "{\\"name\\":\\"foo\\",\\"sellIn\\":9,\\"quality\\":48}",
  "foo,10,50": "{\\"name\\":\\"foo\\",\\"sellIn\\":9,\\"quality\\":49}",
  "foo,11,0": "{\\"name\\":\\"foo\\",\\"sellIn\\":10,\\"quality\\":0}",
  "foo,11,1": "{\\"name\\":\\"foo\\",\\"sellIn\\":10,\\"quality\\":0}",
  "foo,11,49": "{\\"name\\":\\"foo\\",\\"sellIn\\":10,\\"quality\\":48}",
  "foo,11,50": "{\\"name\\":\\"foo\\",\\"sellIn\\":10,\\"quality\\":49}",
  "foo,12,0": "{\\"name\\":\\"foo\\",\\"sellIn\\":11,\\"quality\\":0}",
  "foo,12,1": "{\\"name\\":\\"foo\\",\\"sellIn\\":11,\\"quality\\":0}",
  "foo,12,49": "{\\"name\\":\\"foo\\",\\"sellIn\\":11,\\"quality\\":48}",
  "foo,12,50": "{\\"name\\":\\"foo\\",\\"sellIn\\":11,\\"quality\\":49}",
  "foo,5,0": "{\\"name\\":\\"foo\\",\\"sellIn\\":4,\\"quality\\":0}",
  "foo,5,1": "{\\"name\\":\\"foo\\",\\"sellIn\\":4,\\"quality\\":0}",
  "foo,5,49": "{\\"name\\":\\"foo\\",\\"sellIn\\":4,\\"quality\\":48}",
  "foo,5,50": "{\\"name\\":\\"foo\\",\\"sellIn\\":4,\\"quality\\":49}",
  "foo,6,0": "{\\"name\\":\\"foo\\",\\"sellIn\\":5,\\"quality\\":0}",
  "foo,6,1": "{\\"name\\":\\"foo\\",\\"sellIn\\":5,\\"quality\\":0}",
  "foo,6,49": "{\\"name\\":\\"foo\\",\\"sellIn\\":5,\\"quality\\":48}",
  "foo,6,50": "{\\"name\\":\\"foo\\",\\"sellIn\\":5,\\"quality\\":49}",
  "foo,7,0": "{\\"name\\":\\"foo\\",\\"sellIn\\":6,\\"quality\\":0}",
  "foo,7,1": "{\\"name\\":\\"foo\\",\\"sellIn\\":6,\\"quality\\":0}",
  "foo,7,49": "{\\"name\\":\\"foo\\",\\"sellIn\\":6,\\"quality\\":48}",
  "foo,7,50": "{\\"name\\":\\"foo\\",\\"sellIn\\":6,\\"quality\\":49}",
}
`;
